AQ-28763 Tweaked maintenance mode detection

BenjaminLeitch-AI opened the pull request:
Restored Client's ability to detect maintenance mode after creation,
kept the updated logic. Added unit tests for detection both on and after creation.

BenjaminLeitch-AI committed:
AQ-28763 Tweaked maintenance mode detection

Restored Client's ability to detect maintenance mode after creation,
kept the updated logic. Added unit tests for detection on creation, and
detection after creation.

jessemcdowell-AI commented on src/Aquarius.Client/Samples/Client/WebServiceExceptionHandler.cs:
```diff
@@ -9,6 +9,9 @@ public static class WebServiceExceptionHandler
         public static SamplesApiException CreateSamplesApiExceptionFromResponse(WebServiceException e)
         {
+            if (e.IsAny500())
+                return new SamplesMaintenanceModeException($"{e.Message}: AQUARIUS Samples is in maintenance mode", e);
```
You cannot assume that Any 500 from the server means maintenance mode. This would incorrectly report maintenance for any number of bugs.

The best compromise I can think of is to make a call to the status api in this case, and if it fails report maintenance. You'll need a bit more logic in this function to prevent infinite recursion (don't call status if the error is already for the status api).

jessemcdowell-AI commented on src/Aquarius.Client.UnitTests/Samples/Client/MaintenanceModeTests.cs:
```diff
@@ -0,0 +1,73 @@
+namespace Aquarius.Client.UnitTests.Samples.Client
+{
+    [TestFixture]
+    public class MaintenanceModeTests
```
💯

jessemcdowell-AI reviewed (changes requested):
I think this is a better way to handle the problem, but I expect numerous false-positives for maintenance.

BenjaminLeitch-AI committed:
AQ-28763 Updated conditions for maintenance mode

The client throws a maintenance mode exception iff /status throws a
WebServiceException with a 500 code. If the client catches a
WebServiceException from another endpoint, it will call the /status
endpoint to check for maintenance. Updated the unit tests to reflect
this behavior.
